packages/ra-ui-materialui/src/list/pagination/PaginationActions.tsx

```diff
@@ -1,177 +1,67 @@
-    };
-
-    const renderPageNums = () => {
-        return range().map((pageNum, index) =>
-            pageNum === '.' ? (
-                <span
-                    key={`hyphen_${index}`}
-                    className={PaginationActionsClasses.hellip}
-                >
-                    &hellip;
-                </span>
-            ) : (
-                <Button
-                    size={size}
-                    className={classnames(
-                        'page-number',
-                        PaginationActionsClasses.button,
-                        {
-                            [PaginationActionsClasses.currentPageButton]:
-                                pageNum === page + 1,
-                        }
-                    )}
-                    color={color}
-                    variant={pageNum === page + 1 ? 'outlined' : 'text'}
-                    key={pageNum}
-                    data-page={pageNum - 1}
-                    onClick={gotoPage}
-                >
-                    {pageNum}
-                </Button>
-            )
-        );
+        return translate(`ra.navigation.${type}`, { _: `Go to ${type} page` });
     };
 
-    const nbPages = getNbPages();
-
-    if (nbPages === 1) {
-        return <Root className={PaginationActionsClasses.actions} />;
-    }
-
     return (
-        <Root className={PaginationActionsClasses.actions}>
-            {page > 0 && (
-                <Button
-                    color={color}
-                    size={size}
-                    key="prev"
-                    onClick={prevPage}
-                    className="previous-page"
-                >
-                    {theme.direction === 'rtl' ? (
-                        <ChevronRight />
-                    ) : (
-                        <ChevronLeft />
-                    )}
-                    {translate('ra.navigation.prev')}
-                </Button>
-            )}
-            {renderPageNums()}
-            {page !== nbPages - 1 && (
-                <Button
-                    color={color}
-                    size={size}
-                    key="next"
-                    onClick={nextPage}
-                    className="next-page"
-                >
-                    {translate('ra.navigation.next')}
-                    {theme.direction === 'rtl' ? (
-                        <ChevronLeft />
-                    ) : (
-                        <ChevronRight />
-                    )}
-                </Button>
-            )}
+        <Root>
+            <Pagination
+                size={size}
+                count={nbPages}
+                // <TablePagination>, the parent, uses 0-based pagination
+                // while <Pagination> uses 1-based pagination
+                page={page + 1}
+                onChange={(e: any, page) => onPageChange(e, page - 1)}
+                {...sanitizeRestProps(rest)}
+                getItemAriaLabel={getItemAriaLabel}
```

Comment:
`getItemAriaLabel` cannot be overridden

Reply:
that's on purpose. The parent, TablePagination, passes its own getItemAriaLabel function, which is compatible with TablePaginationActions, but not with Pagination (the two signatures are different).